### ISSUE TITLE
False-positive `PossiblyNullArgument` when using `??=` on `ArrayAccess` with nullable `offsetGet()`

### ISSUE THREAD

**Issue by Shira-3749** [vimeo/psalm#10857]
When using `??=` on an implementation of `ArrayAccess` whose `offsetGet()` can return `null` for nonexistent offsets, I'm getting a false-positive `PossiblyNullArgument`:

https://psalm.dev/r/f82e852479

```php
// Psalm infers $foo as Foo|null even though that's impossible
$foo = ($map['foo'] ??= new Foo());
```

The  `offsetGet()` method is not called at all when `??=` is used so it is impossible for it to influence the call to `offsetSet()` or the result of the `??=` operator:

https://3v4l.org/vQqLV

**Comment by jacekkarczmarczyk** [comment]
Duplicate of https://github.com/vimeo/psalm/issues/9960?

**Comment by Shira-3749** [comment]
It could be same cause, just with `??=` instead of `??`.
